### PR TITLE
Dvbcsa

### PR DESCRIPTION
**Pull request by quarium**

### CODE REVIEW

File: include/upipe-dvbcsa/upipe_dvbcsa_bs_decrypt.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef _UPIPE_DVBCSA_UPIPE_DVBCSA_BS_DEC_H_
```
Comment (cmassiot) [inline include/upipe-dvbcsa/upipe_dvbcsa_bs_decrypt.h]: Should be _UPIPE_DVBCSA_UPIPE_DVBCSA_BS_DECRYPT_H_

File: include/upipe-dvbcsa/upipe_dvbcsa_common.h [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    /** set the maximum latency (uint64_t) */
+    UPIPE_DVBCSA_COMMON_SET_MAX_LATENCY,
+
+    /** add a pid to the encryption/decryption list (uint64_t) */
+    UPIPE_DVBCSA_COMMON_ADD_PID,
+    /** delete a pid from the encryption/decryption list (uint64_t) */
+    UPIPE_DVBCSA_COMMON_DEL_PID,
+
+    /** custom dvbcsa commands start here */
+    UPIPE_DVBCSA_CONTROL_LOCAL,
```
Comment (cmassiot) [inline include/upipe-dvbcsa/upipe_dvbcsa_common.h]: Either DVBCSA_COMMON everywhere, or DVBCSA everywhere (fine by me).

File: include/upipe-dvbcsa/upipe_dvbcsa_bs_decrypt.h (hunk 1 above)
Comment (quarium) [reply include/upipe-dvbcsa/upipe_dvbcsa_bs_decrypt.h]: FIXED

File: lib/upipe-dvbcsa/upipe_dvbcsa_bs_decrypt.c [hunk 3]
```diff
@@ -0,0 +1,527 @@
...
+
+#include <bitstream/mpeg/ts.h>
+#include <dvbcsa/dvbcsa.h>
+
+#include "common.h"
+
+/** expected input flow format */
+#define EXPECTED_FLOW_DEF "block.mpegts."
+/** dvbcsa decrypt latency */
+#define DVBCSA_LATENCY  (UCLOCK_FREQ / 100)
```
Comment (cmassiot) [inline lib/upipe-dvbcsa/upipe_dvbcsa_bs_decrypt.c]: Add a comment to say this value comes from a statistic on normal hardware.

File: include/upipe-dvbcsa/upipe_dvbcsa_common.h (hunk 2 above)
Comment (quarium) [reply include/upipe-dvbcsa/upipe_dvbcsa_common.h]: Fixed

File: lib/upipe-dvbcsa/upipe_dvbcsa_bs_decrypt.c (hunk 3 above)
Comment (quarium) [reply lib/upipe-dvbcsa/upipe_dvbcsa_bs_decrypt.c]: Fixed

**Pull request merged**